js/whitebit.js

```diff
@@ -943,6 +952,12 @@ module.exports = class whitebit extends Exchange {
         const stopPrice = this.safeString (order, 'activation_price');
         const orderId = this.safeString2 (order, 'orderId', 'id');
         const type = this.safeString (order, 'type');
+        let amount = this.safeString (order, 'amount');
+        const createMarketBuyOrderRequiresPrice = this.safeValue (this.options, 'createMarketBuyOrderRequiresPrice', true);
+        if (createMarketBuyOrderRequiresPrice && side === 'buy' && type.indexOf ('limit') < 0) {
+            // in these cases the amount is in the quote currency
+            amount = Precise.stringDiv (amount, price);
```

Comment:
@kroitor since this is an `createMarketBuyOrderRequiresPrice` exchange, we need to make this conversion for buy-market orders,  right?

Reply:
I don't know why yet but the API is returning 0 for the price in buy orders at least, I'll probably try to reach out the support again, or I can't confirm this conversion is working properly
![image](https://user-images.githubusercontent.com/43336371/146845634-dd41da62-c90d-40db-8fef-d049f31f4851.png)

Reply:
In this case, we should not look into `createMarketBuyOrderRequiresPrice` – that lookup is only done for the warning, t should look like similar to these examples:

- https://github.com/ccxt/ccxt/blob/master/js/binance.js#L2482
- https://github.com/ccxt/ccxt/blob/master/js/okex.js#L1697
- https://github.com/ccxt/ccxt/blob/master/js/huobi.js#L2387

Documented here:

- https://github.com/ccxt/ccxt/wiki/Manual#market-buys

Let me know if that is not entirely clear.

Reply:
@kroitor I'm not sure I get it since this is in the parseOrder and the examples you gave are in the createOrder 🤔

Reply:
@carlosmiei i think in this case createOrder produces a 0-price response that lands into parseOrder, but the initial problem originates in createOrder. Correct me if i'm wrong.

Reply:
We should not do `stringDiv` there, cause cost = amount * price, so it's either `stringMul` or something else.

Reply:
Actually, on a second thought, my two previous comments may be wrong. Need to investigate.

Reply:
If the `price` in their response is zero, we set it to `undefined` in general.

Reply:
@kroitor  No, the problem is not in the createOrder, the zero price problem is in the API itself. I've confirmed it with support and hopefully will be fixed soon.

I was just asking if the "conversion logic" from the amount in base currency to the default (quote currency ) was ok.
since cost = amount * price, actually, I'm doing amount = cost/price, so I'd say it matches.
![image](https://user-images.githubusercontent.com/43336371/147011851-8b1b7943-9552-4b2c-bd48-2c7d989e7d51.png)

Reply:
If it's a market order, let's set price = `undefined`.

Reply:
@kroitor and what about the amount, should we keep as is, in the base quote for market-buy orders? (for instance in USDT for a BTC/USDT order)

Reply:
We should determine which currency it is from the other fields or from the website, and either fill `amount` or `cost`, but probably not both.

Reply:
I've changed it accordingly